src/mlpack/tests/main_tests/logistic_regression_test.cpp

```diff
@@ -0,0 +1,378 @@
+    	Log::Fatal << "Invalid Test Name : " << e.what() << std::endl;
+    }
+  
+  }
+
+  ~LogisticRegressionTestFixture()
+  {
+    // Clear the settings.
+    CLI::ClearSettings();
+  }
+
+};
+
+BOOST_FIXTURE_TEST_SUITE(LogisticRegressionMainTest, LogisticRegressionTestFixture);
+
+/** 
+  * Ensuring that absence of training data is checked.
+ **/
+BOOST_AUTO_TEST_CASE(LRNoTrainingData) 
+{
+  arma::Row<size_t> trainY;
+  trainY << 0 << 1 << 0 << 1 << 1 << 1 << 0 << 1 << 0 << 0 << arma::endr; // 10 responses
+  
+  SetInputParam("labels", std::move(trainY));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+}
+
+/**
+ * Ensuring that absence of responses is checked.
+ */
+BOOST_AUTO_TEST_CASE(LRNoResponses)
+{
+  constexpr int N = 10;
+  constexpr int D = 1;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  SetInputParam("training", std::move(trainX));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking that that size and dimensionality of prediction is correct.
+ */
+BOOST_AUTO_TEST_CASE(LRPridictionSizeCheck)
+{
+  constexpr int N = 10;
+  constexpr int D = 3;
+  constexpr int M = 15;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  arma::Row<size_t> trainY;
+  trainY << 0 << 1 << 0 << 1 << 1 << 1 << 0 << 1 << 0 << 0 << arma::endr; // 10 responses
+  arma::mat testX = arma::randu<arma::mat>(D, M);
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+  SetInputParam("test", std::move(testX));
+
+  mlpackMain();
+
+  const arma::Row<size_t> testY = CLI::GetParam<arma::Row<size_t>>("output");
+
+  BOOST_REQUIRE_EQUAL(testY.n_rows, 1);
+  BOOST_REQUIRE_EQUAL(testY.n_cols, M);
+}
+
+/**
+  * Ensuring that the response size is checked
+ **/
+BOOST_AUTO_TEST_CASE(LRWrongResponseSizeTest) 
+{
+  constexpr int D = 3;
+  constexpr int N = 10;
+  
+  arma::mat trainX = arma::randu<arma::mat>(D,N);
+  arma::Row<size_t> trainY; // response vector with wrong size
+
+  trainY << 0 << 0 << 1 << 0 << 1 << 1 << 1 << 0 << arma::endr; // 8 responses - incorrect size
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(),std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking two options of specifying responses (extra row in train matrix and
+ * extra parameter) and ensuring that predictions are the same.
+ */
+BOOST_AUTO_TEST_CASE(LRResponsesRepresentationTest)
+{
+  arma::mat trainX1({{1.0, 2.0, 3.0}, {1.0, 4.0, 9.0}, {0,1,1}});
+  arma::mat testX({{4.0,5.0},{1.0,6.0}});
+
+  SetInputParam("training",std::move(trainX1));
+  SetInputParam("test", testX);
+
+  // The first solution.
+  mlpackMain();
+
+  const arma::Row<size_t> testY1 = CLI::GetParam<arma::Row<size_t>>("output");
```

Comment:
Here I would suggest using `std::move()`, but again, since these objects are so small for the tests, it doesn't matter too much.  But if you were writing other mlpack code where the data was larger I would be more insistent. :)

Reply:
Done. I made the changes accordingly. Thanks for your suggestion. I will remember that next time.